runtimes/web/src/framebuffer.js

```diff
@@ -29,7 +39,64 @@ export class Framebuffer {
         }
     }
 
-    drawRect (x, y, width, height) {
+    drawHLineInternal(color, x, y, len) {
+        if (y > HEIGHT || y < 0 || len === 0) {
+            return;
+        }
+        let startX, endX;
+
+        if(len < 0) {
+            startX = Math.max(0, x + len + 1);
+            endX = Math.min(WIDTH, x + 1);
+        } else {
+            startX = Math.max(0, x);
+            endX = Math.min(x + len, WIDTH);
+        }
+
+        for (let xx = startX; xx < endX; ++xx) {
+            this.drawPoint(color, xx, y);
+        }
+    }
+
+    drawVLineInternal(color, x, y, len) {
+        if (x > WIDTH || x < 0 || len === 0) {
+            return;
+        }
+
+        let startY,endY;
+
+        if (len < 0) {
+            startY = Math.max(0, y + len + 1);
+            endY = Math.min(HEIGHT, y + 1);
+        } else {
+            startY = Math.max(0, y);
+            endY = Math.min(HEIGHT, y + len);
+        }
+
+        for (let yy = startY; yy < endY; ++yy) {
+            this.drawPoint(color, x, yy);
+        }
+    }
+
+    drawHLine(x, y, len) {
+        const strokeColor = getStrokeColor(this.drawColors);
+
+        if(strokeColor === 0) {
+            return;
+        }
+        this.drawHLineInternal(strokeColor, x, y, len);
+    }
+
+    drawVLine(x, y, len) {
+        const strokeColor = getStrokeColor(this.drawColors);
+
+        if(strokeColor === 0) {
+            return;
+        }
+        this.drawVLineInternal(strokeColor, x, y, len);
+    }
```

Comment:
Why not just inline `drawVLineInternal` and `drawHLineInternal` into `drawHLine`/`drawVLine`?

Reply:
I was wandering if we could reuse those internal functions to fill rect and oval.

It on the `Todo` list to verify this possibility.

See: https://github.com/aduros/wasm4/issues/74#issuecomment-917483818

It makes sense to inline them if we are not going to use them.

I'd like a bit of feedback on this topic.

Reply:
I see. In this case make sense hoist all checks as well out of internal methods and use only loop like:
```ts
function drawHLineInternal(xstart, xend, y) {
   for (let xx = xstart, xx < xend, xx++) {
      this.drawPoint(color, xx, y);
   }
}
```

Reply:
You're right.
I'll  probably use [typed-array.prototype.fill tho](https://github.com/aduros/wasm4/pull/103#issuecomment-922972578)